pkg/cnab/cnab-to-oci/registry.go

```diff
@@ -188,3 +189,30 @@ func (r *Registry) getDockerClient() (*command.DockerCli, error) {
 	}
 	return cli, nil
 }
+
+func (r *Registry) IsInvocationImageExists(invocationImage string) (bool, error) {
+	ctx := context.Background()
+
+	cli, err := r.getDockerClient()
+	if err != nil {
+		return false, err
+	}
+
+	ref, err := cnab.ParseOCIReference(invocationImage)
+	if err != nil {
+		return false, err
+	}
+
+	imageListOpts := types.ImageListOptions{All: true, Filters: filters.NewArgs(filters.KeyValuePair{Key: "reference", Value: ref.Repository()})}
+
+	imageSummaries, err := cli.Client().ImageList(ctx, imageListOpts)
+	if err != nil {
+		return false, err
```

Comment:
Here's a bit of background on how we are handling errors in Porter. If the error came from a third party library or the stdlibrary (as opposed to Porter's code), it really helps to wrap the error and provide context about what action was being attempted with any supplemental information (like the name of the image in this case). That way we avoid printing errors to the console like "Error: file not found" or "Error: EOF" without context about what was going on. 😀 

Otherwise, if the function you called is Porter's code, do not wrap the error, because we already wrapped the error at the source and provided context. There's no need to re-wrap the error in most cases.

So in this case, the error here should be wrapped, and then in stamp.go line 67, you would not wrap the error and just return it as-is.

Let me know if you have questions about that, I realize it's a bit of a special thing that Porter does (and not always consistently like we should).

Reply:
This was a really good review, a new thing that I learned. Yep, the wrapping scenario makes sense to be done and I can see how it is a good practice to do so. Thank you for the really thorough explanation, Carolyn! 🙌